locator/token_metadata.cc

```diff
@@ -420,7 +420,7 @@ future<> token_metadata_impl::update_normal_tokens(std::unordered_set<token> tok
         co_return;
     }
 
-    if (!_topology.has_endpoint(endpoint, topology::pending::no)) {
+    if (!_topology.has_endpoint(endpoint)) {
```

Comment:
Doesn't this change now? We'll get pending endpoints too.

Reply:
The function, update_normal_tokens, is the one applying the token changes once an endpoint stopped being pending and joins the ring as normal token owner.
It already relied on the caller invoking update_topology prior to calling update_normal_tokens so the endpoint be found here (with topology::pending::no), otherwise -> crash.

So in today's state of the code, an endpoint being in topology is orthogonal to its state in token_metadata.
(One is_member changed to rely on the normal_token_owner state rather than the topology pending state,
in a44ca0690642b0b33d78932432fbab92de1cc0f7)